llvm/include/llvm/Support/GenericDomTreeConstruction.h

```diff
@@ -123,13 +132,29 @@ struct SemiNCAInfo {
     return Res;
   }
 
-  NodePtr getIDom(NodePtr BB) const {
-    auto InfoIt = NodeToInfo.find(BB);
-    if (InfoIt == NodeToInfo.end()) return nullptr;
+  template <class T_ = NodeT>
```

Comment:
Prefer `typename` and other letters `U`, `P`, etc. instead of adding underscores. This is for consistency with the surrounding code.

Reply:
Used if constexpr instead, as in #101705.